packages/hardhat-core/src/internal/cli/project-creation.ts

```diff
@@ -88,11 +121,24 @@ async function printWelcomeMessage() {
   );
 }
 
-async function copySampleProject(projectRoot: string) {
+async function copySampleProject(
+  projectRoot: string,
+  projectType: SampleProjectTypeCreationAction
+) {
   const packageRoot = getPackageRoot();
 
   await fsExtra.ensureDir(projectRoot);
-  await fsExtra.copy(path.join(packageRoot, "sample-project"), projectRoot);
+  await fsExtra.copy(
+    path.join(packageRoot, "sample-projects", "basic"),
+    projectRoot
+  );
+
+  if (projectType === Action.CREATE_ADVANCED_SAMPLE_PROJECT_ACTION) {
```

Comment:
Let's add a comment here explaining that the advanced project _needs_ the basic projects files. Otherwise, the lack of an `if` statement for the previous `fs.copy` is a bit surprising.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1632/commits/49d6c68d37354de46eabc58bc54f6402529110a0